samples/MerQure.Tools.Samples/RetryBusExample/Domain/ActionService.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MerQure.Tools.Samples.RetryBusExample.Domain
+{
+    public class ActionService
+    {
+        private static int _fakeError;
+
+        private readonly ISampleService _sampleService;
+
+        public ActionService(ISampleService sampleService)
+        {
+            _sampleService = sampleService;
+        }
+
+        public void SendNewSample()
+        {
+            _sampleService.Send(new Sample()
+            {
+                Name = "MerQure Tools"
+            });
+        }
+
+        public void Consume()
+        {
+            _sampleService.Consume(async (object sender, Sample sample) => await OnSampleReceived(sender, sample));
+        }
+
+        public async Task OnSampleReceived(object sender, Sample sample)
+        {
+            try
+            {
+                await ManageNewSample(sample);
+            }
+            catch (SampleException)
+            {
+                _sampleService.RetryLater(sample);
+            }
+            catch (Exception e)
+            {
+                _sampleService.SendOnError(sample);
+                //...
+            }
+        }
+
+        private async Task ManageNewSample(Sample sample)
+        {
+            _fakeError++;
+
+            int value = new Random().Next(1000, 3000);
+            await Task.Delay(value);
+            if (_fakeError % 15 == 0)
+            {
+                throw new Exception("This is an unmanageable exception, like NPE");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS112)